### ISSUE TITLE
TensorFlow wrapper ravels outputs on predict

### ISSUE THREAD

**Issue by dwyatte** [mlflow/mlflow#1066]
The TensorFlow wrapper ravels outputs when predict is called: https://github.com/mlflow/mlflow/blob/master/mlflow/tensorflow.py#L286

This makes the wrapper incompatible with models that produce multiple outputs (e.g., Softmax) when called with Spark UDFs, which expect the index size to be preserved.

Perhaps it might be better to return a DataFrame with columns:
[output_tensor_name_0, output_tensor_name_1, ..., output_tensor_name_N].

**Comment by tomasatdatabricks** [comment]
Hey @dwyatte , thanks for creating the issue! 
Can we just return the multidimensional array as is?

**Comment by andychow-db** [comment]
This seems like a very useful thing to have.
